cores/esp8266/lwIPIntf.h

```diff
@@ -0,0 +1,28 @@
+
+#ifndef _LWIPINTF_H
+#define _LWIPINTF_H
```

Comment:
`lwIPIntfDev` is the device, inheriting from raw device `RawDevice`, and `lwIPIntf`. That one contains common methods for any interface, including the arduino reordering helper because any interface can be a dhcp server (except STA because we want to). This class contains `hostname()` too.